util/zenfs.cc

```diff
@@ -154,20 +160,22 @@ int zenfs_tool_list() {
 
 int zenfs_tool_df() {
   Status s;
-  std::unique_ptr<ZonedBlockDevice> zbd = zbd_open(FLAGS_zbd, true, false);
-  if (!zbd) return 1;
-  ZonedBlockDevice *zbdRaw = zbd.get();
-
+  std::unique_ptr<ZonedBlockDevice> zbd;
   std::unique_ptr<ZenFS> zenFS;
-  s = zenfs_mount(zbd, &zenFS, true);
+
+  s = zenfs_zbd_open(FLAGS_zbd, true, false, zbd);
+  if (!s.ok() || !zbd) return 1;
+
+  uint64_t used = zbd->GetUsedSpace();
+  uint64_t free = zbd->GetFreeSpace();
+  uint64_t reclaimable = zbd->GetReclaimableSpace();
```

Comment:
Is there a reason why `used`, `free` and `reclaimable` are now assigned before checking for a successful mount?

Reply:
Yes, once you pass the zbd too ZenFS, you cannot call it. Holding on to the raw pointer of a unique_ptr defeats the purpose.